linetools/isgm/utils.py

```diff
@@ -451,4 +451,81 @@ def coincident_components(comp1, comp2, tol=0.2*u.arcsec):
             overlap = line1.coincident_line(line2)
             if overlap is True:
                 return True
-    return False
\ No newline at end of file
+    return False
+
+
+def group_coincident_compoments(comp_list):
+    """For a given input list of component, this function
+    group together components that are coincident to each other
+    (including by transitivity), and return them as a list of
+    component lists.
+
+    Parameters
+    ----------
+    comp_list : list of AbsComponent
+        Input list of components to group
+
+    Returns
+    -------
+    output : list of lists of AbsComponent
```

Comment:
I thought about it, but I could not come with a sample naming convention for the keys that make sense. Remember that these are components being coincident in wobs space that may or may not be physically associated; mean redshift will be misleading. Because of the lack of a good naming for the keys, I decided to go for an output in list form rather than dictionary.